byterun/caml/mlvalues.h

```diff
@@ -186,7 +183,24 @@ bits  63        (64-P) (63-P)        10 9     8 7   0
 /* Pointer to the first field. */
 #define Op_val(x) ((value *) (x))
 /* Fields are numbered from 0. */
+#if CAML_API_VERSION < 405
 #define Field(x, i) (((value *)(x)) [i])           /* Also an l-value. */
+#endif
+
+static inline void caml_read_field (value obj, mlsize_t f, value* ret)
+{
+  *ret = Op_val(obj)[f];
+}
+
+static inline value Field_imm (value obj, mlsize_t f)
```

Comment:
`Field_immut` is a better name, I hadn't thought of the confusion with "immediate".

`Immut_field` may read better, but it's inconsistent with the rest of the API, where `Foo_bar` consistently means "the foo of/in the bar" (e.g. `Wosize_val`, `Double_field`). With this convention, `Immut_field` sounds like you're reading an immutable object out of a possibly-mutable field, which is wrong: this function is for accessing immutable fields, no matter what sort of data they contain.